File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        if (nums.length < 2) return false;
        
        Map<Integer, Integer> mp1 = new HashMap<>();
        long sum = 0;
        
        mp1.put(0, -1); // To handle cases where the subarray starts from the beginning
        
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
            int mod = (int) (sum % k);
            
            if (mp1.containsKey(mod)) {
                if (i - mp1.get(mod) > 1) return true;
            } else {
                mp1.put(mod, i);
            }
        }
        
        return false;
    }
}

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
          unordered_map<int,int>mp1;
          long long sum=0;
        if(nums.size()<2) return 0;
        mp1[0]=-1;
        for(int i=0;i<nums.size();i++)
        {
        
            sum+=nums[i];
            if(mp1.find(sum%k)!=mp1.end())
            {
              if(i-mp1[sum%k]>1)return 1;  
            }
            else
                mp1[sum%k]=i;
        }
        return 0;
    }
};